### ISSUE TITLE
MultiSelect (select all option): design and a11y review

### ISSUE THREAD

**Issue by laurenmrice** [carbon-design-system/carbon#13612]
## Acceptance criteria
- From the research findings and iterations that came out of [#13611](https://github.com/carbon-design-system/carbon/issues/13611), playback to the design team and accessibility for feedback.

**Playback 1**
- Get feedback from design and accessibility in a Design crit meeting.

**Update designs with feedback**
- Implement feedback in designs from the design and accessibility playback.

**Playback 2**
- Playback designs again to the team to land on a final design solution.

**Comment by tay1orjones** [comment]
If the design explorations land on the "Select all" to be just another item in the list, here's a couple of my assumptions to review with CAG:

1. For the "Select all" to be accessible, that checkbox will need to have `aria-controls` with a value of a string containing the `id`s of every other item in the list.
1. If "Select all" is selected, and then one or more other items are deselected, the "Select all" checkbox should have `aria-checked="mixed"` and styled to support an indeterminate state as is done in other components like DataTable
